File: Maximum Depth of N-ary Tree/Maximum_Depth_of_N-ary_Tree.py
# Runtime: 664 ms, faster than 32.94% of Python3 online submissions for Maximum Depth of N-ary Tree.
# Memory Usage: 95.3 MB, less than 8.33% of Python3 online submissions for Maximum Depth of N-ary Tree.

"""
# Definition for a Node.
class Node:
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""
class Solution:
    def maxDepth(self, root: 'Node') -> int:
        if root is None:
            return 0
        
        res = 0
        for children in root.children:
            res = max(res, self.maxDepth(children))
        
        return res + 1

File: Maximum Depth of N-ary Tree/Maximum_Depth_of_N-ary_Tree.cpp
// Runtime: 140 ms, faster than 80.18% of C++ online submissions for Maximum Depth of N-ary Tree.
// Memory Usage: 32 MB, less than 90.00% of C++ online submissions for Maximum Depth of N-ary Tree.

/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/
class Solution 
{
public:
    int maxDepth(Node* root) 
    {
        if (root == nullptr)
            return 0;

        int res = 0;
        for (auto children : root->children)
        {
            res = max(res, maxDepth(children));
        }    
        
        return res + 1;
    }
};